bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1128 @@
+            Channel targetChannel = thing.getChannel(channelId);
+            if (Objects.nonNull(dynamicsChannel) && Objects.nonNull(targetChannel)) {
+                logger.debug("{} - handleDynamicsCommand() channelId:{}, command:{}, duration:{}", resourceId,
+                        channelId, command, duration);
+                handleCommand(dynamicsChannel.getUID(), duration);
+                handleCommand(targetChannel.getUID(), command);
+                return;
+            }
+        }
+        logger.warn("{} - handleDynamicsCommand() failed for channelId:{}, command:{}, duration:{}", resourceId,
+                channelId, command, duration);
+    }
+
+    @Override
+    public void initialize() {
+        Clip2ThingConfig config = getConfigAs(Clip2ThingConfig.class);
+
+        String resourceId = config.resourceId;
+        if (Objects.isNull(resourceId) || resourceId.isEmpty()) {
+            logger.debug("{} -> initialize() configuration resourceId is bad", resourceId);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.api2.conf-error-resource-id-bad");
+            return;
+        }
+        thisResource.setId(resourceId);
+        this.resourceId = resourceId;
+        logger.debug("{} -> initialize()", resourceId);
+
+        updateThingFromLegacy();
+        updateStatus(ThingStatus.UNKNOWN);
+
+        dynamicsDuration = Duration.ZERO;
+        dynamicsExpireTime = Instant.MIN;
+
+        disposing = false;
+        hasConnectivityIssue = false;
+        updatePropertiesDone = false;
+        updateDependenciesDone = false;
+        updateLightPropertiesDone = false;
+        updateSceneContributorsDone = false;
+
+        Bridge bridge = getBridge();
+        if (Objects.nonNull(bridge)) {
+            BridgeHandler bridgeHandler = bridge.getHandler();
+            if (bridgeHandler instanceof Clip2BridgeHandler) {
+                ((Clip2BridgeHandler) bridgeHandler).childInitialized();
+            }
+        }
+    }
+
+    /**
+     * Update the channel state depending on a new resource sent from the bridge.
+     *
+     * @param resource a Resource object containing the new state.
+     */
+    public void onResource(Resource resource) {
+        if (!disposing) {
+            boolean resourceConsumed = false;
+            String incomingResourceId = resource.getId();
+            if (resourceId.equals(incomingResourceId)) {
+                if (resource.hasFullState()) {
+                    thisResource = resource;
+                    if (!updatePropertiesDone) {
+                        updateProperties(resource);
+                        resourceConsumed = updatePropertiesDone;
+                    }
+                }
+                if (!updateDependenciesDone) {
+                    resourceConsumed = true;
+                    threadPool.execute(() -> updateDependencies());
```

Comment:
This task needs to be cancelled in `dispose()`.

Reply:
See my comment elsewhere about the quickest way to terminate a task that will have been started immediately (via `submit()` or `excute()`)

Reply:
Sorry, this particular comment was by mistake.